data.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type Data struct {
+	Project string `json:"project"`
+	Distro  string `json:"distro"`
+	Kernel  string `json:"kernel"`
+	Flatpak bool   `json:"flatpak"`
+	AVX     bool   `json:"avx"`
+	CPU     string `json:"cpu"`
+	GPUs    string `json:"gpu"` // list seperated by commas
+}
+
+var (
+	ErrDataBadLength             = errors.New("data member has invalid length")
+	ErrDataFlatpakDistroMismatch = errors.New("flatpak distro has mismatch")
+	ErrDataBadGPUs               = errors.New("gpus given is invalid")
+	ErrDataBadCPUVendor          = errors.New("cpu given has bad vendor")
+	ErrDataBadCPULength          = errors.New("cpu given has bad length")
+)
+
+// Generate the CSV Header based on the JSON tags in the Data struct at run-time
+var CSVHeader = func() (h []string) {
+	var d Data
+	t := reflect.TypeOf(d)
+
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		h = append(h, f.Tag.Get("json"))
+	}
+
+	return
+}()
+
+func (d Data) CSV() []string {
+	return []string{
+		d.Project, d.Distro, d.Kernel,
+		strconv.FormatBool(d.AVX), strconv.FormatBool(d.Flatpak),
+		d.CPU, d.GPUs,
+	}
+}
+
+func (d *Data) Sanitize() {
+	reg := regexp.MustCompile(`[^a-zA-Z0-9@*().\-,\s]+`)
```

Comment:
Regex is too wide, you need to disable \s in favor for just a space character.

Reply:
Provide a new regex which applies to all fields.

Reply:
`[^a-zA-Z0-9@*().\- ]+`